src/docfx/restore/RestoreGit.cs

```diff
@@ -142,9 +143,13 @@ void AddWorkTrees()
                                         // clean existing work tree folder
                                         // it may be dirty caused by last failed restore action
                                         if (Directory.Exists(workTreePath))
+                                        {
+                                            // https://stackoverflow.com/questions/24265481/after-directory-delete-the-directory-exists-returning-true-sometimes
                                             Directory.Delete(workTreePath, true);
+                                            while (Directory.Exists(workTreePath))
+                                                Thread.Sleep(1);
```

Comment:
this can hang forever if some other process holds a lock.

Reply:
you are right, but you can see that work tree path is hidden from end user, controlled by docfx, so I don't think it's a really big problem, but I can add a timeout here or find another way to make sure it's deleted.

Reply:
@yufeih  and @herohua , no actually, because it the work tree path is using by other process, the Directory.Delete will throw IO exceptions

![image](https://user-images.githubusercontent.com/19990166/57927091-b3238380-78df-11e9-8fb6-cf0a5476ecd1.png)

Reply:
that code actually is just waiting the directory handler to be close after being deleted.